source/common/config/grpc_mux_subscription_impl.cc

```diff
@@ -81,7 +81,14 @@ void GrpcMuxSubscriptionImpl::onConfigUpdateFailed(ConfigUpdateFailureReason rea
     ENVOY_LOG(warn, "gRPC config for {} rejected: {}", type_url_, e->what());
     break;
   }
+
   stats_.update_attempt_.inc();
+  if (Envoy::Config::ConfigUpdateFailureReason::ConnectionFailure == reason) {
+    // New gRPC stream will be established and send requests again.
+    // If init_fetch_timeout is non-zero, server will continue startup after it timeout
+    return;
+  }
+
   callbacks_.onConfigUpdateFailed(reason, e);
```

Comment:
nit: call this method if reason != ConnectionFailure may be more readable than return?

Reply:
Putting const value at left side of `==` is considered a better style. It doesn't cause any difficulty for readability usually, because reader always need to see both side of `==` to understand the condition.

Reply:
Sorry. My comment was not clear may be. I am not talking about const on left of ==,

I am suggesting the following - Not a big deal though
```
if (Envoy::Config::ConfigUpdateFailureReason::ConnectionFailure != reason) {
callbacks_.onConfigUpdateFailed(reason, e);
}
```

Reply:
+1

Reply:
@l8huang can you actually switch this to `reason == the const value`? I agree that what you have there is "better style" for the reason you give, but it's actually quite jarring as most of the Envoy code base doesn't do this, and the "conform to local practices" style argument wins out IMHO.